src/windows.h

```diff
@@ -0,0 +1,46 @@
+#ifndef GMTIME_H
+#define GMTIME_H
+#if defined(WIN32) || defined(_WIN32) || defined(__WIN32) && !defined(__CYGWIN__)
+#pragma comment(lib, "ws2_32.lib")
+#include <stdio.h>
+#include <stdlib.h>
+#define timegm _mkgmtime
+
+#define gmtime_r __gmtime_r
+static inline struct tm *gmtime_r(const time_t *clock, struct tm *result)
+{
+  struct tm *data;
+  if (!clock || !result)
+    return NULL;
+  data = gmtime(clock);
+  if (!data)
+    return NULL;
+  memcpy(result, data, sizeof(*result));
+  return result;
+}
+
+#include <time.h>
+#include <iomanip>
+#include <sstream>
+
+static inline char* strptime(const char* s,
+                          const char* f,
+                          struct tm* tm) {
```

Comment:
Why do you need this? Very ugly.

Reply:
strptime is not available on windows. Is there some better way of doing this?